### PR TITLE
Clean code

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Also Fixes https://github.com/47deg/helios/issues/117

### CODE REVIEW

**Comment by 47degdev** [comment]
***Hood benchmark comparison:***
:heavy_check_mark: `Decoding` (Threshold: 223.10153204380887)

Benchmark | Value
|--|--|
master_benchmark | 11089.726408483377
helios_benchmark | 11308.577991803551


:warning: `Decodingfromraw` (Threshold: 250.0)

Benchmark | Value
|--|--|
master_benchmark | 7307.124761797657
helios_benchmark | 7184.254146800757


:heavy_check_mark: `Parsing` (Threshold: 250.0)

Benchmark | Value
|--|--|
master_benchmark | 22135.769525357944
helios_benchmark | 23100.66795252022

**Comment by AdrianRaFo** [comment]
Could you @nomisRev take a look at this when you have a chance?

File: helios-core/src/main/kotlin/helios/instances/instancesEq.kt [hunk 1]
```diff
@@ -48,38 +47,38 @@ interface JsNumberEqInstance : Eq<JsNumber> {
...
+    is JsInt     -> when (b) {
       is JsDecimal -> String.eq().run { this@eqv.value.toString().eqv(b.value) }
-      is JsLong -> Long.eq().run { this@eqv.value.toLong().eqv(b.value) }
-      is JsDouble -> Double.eq().run { this@eqv.value.toDouble().eqv(b.value) }
-      is JsFloat -> Float.eq().run { this@eqv.value.toFloat().eqv(b.value) }
-      is JsInt -> Int.eq().run { this@eqv.value.eqv(b.value) }
+      is JsLong    -> Long.eq().run { this@eqv.value.toLong().eqv(b.value) }
+      is JsDouble  -> Double.eq().run { this@eqv.value.toDouble().eqv(b.value) }
+      is JsFloat   -> Float.eq().run { this@eqv.value.toFloat().eqv(b.value) }
+      is JsInt     -> Int.eq().run { this@eqv.value.eqv(b.value) }
```
Comment (nomisRev) [inline helios-core/src/main/kotlin/helios/instances/instancesEq.kt]: Is this a setting in KtLint? I like it but it's something I haven't seen in Kotlin but only 

How can I add it in my own projects? :D

File: helios-core/src/test/kotlin/helios/instances/InstancesJavaTimeTest.kt [hunk 2]
```diff
@@ -26,7 +25,9 @@ class InstancesJavaTimeTest : UnitSpec() {
 
     "ZonedDateTime should be encoded and decoded successfully"{
       val now = ZonedDateTime.now()
-      ZonedDateTimeDecoderInstance().decode(ZonedDateTimeEncoderInstance().run { now.encode() }) should beRight(now)
+      ZonedDateTimeDecoderInstance().decode(ZonedDateTimeEncoderInstance().run { now.encode() }) should beRight(
```
Comment (nomisRev) [inline helios-core/src/test/kotlin/helios/instances/InstancesJavaTimeTest.kt]: We should probably split these off in a Java8 module for Android support in a separate ticket? Looks like a great beginner ticket for interested contributors.

File: helios-core/src/main/kotlin/helios/instances/instancesEq.kt (hunk 1 above)
Comment (AdrianRaFo) [reply helios-core/src/main/kotlin/helios/instances/instancesEq.kt]: No idea I think it was the detekt plugin from Intellij

File: helios-core/src/test/kotlin/helios/instances/InstancesJavaTimeTest.kt (hunk 2 above)
Comment (AdrianRaFo) [reply helios-core/src/test/kotlin/helios/instances/InstancesJavaTimeTest.kt]: https://github.com/47deg/helios/issues/126

**Pull request merged**